teku/src/main/java/tech/pegasys/teku/Teku.java

```diff
@@ -35,7 +36,13 @@ public static void main(final String... args) {
   }
 
   private static void start(final TekuConfiguration config) {
-    final BeaconNode node = new BeaconNode(config);
+    final Node node;
+    if (config.isValidatorNodeOnly()) {
+      node = new ValidatorNode(config);
+    } else {
+      node = new BeaconNode(config);
+    }
```

Comment:
Somewhere along the way we probably should support running the beacon node without it starting an internal validator service. Maybe not step one though.